### PR TITLE
unit 2

### PR DESCRIPTION
**Pull request by DanielCalimayor**

### CODE REVIEW

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
# making_forecasts - "Daniel Calimayor-Montero"

# Part 1 - Loading Data
***
**What went well:**

- Good job with importing the provided tools and implementing a good (maybe complex) way of taking in the metrics data from Wikipedia.:

```python

u = pd.read_csv('pageviews-20150701-20190523.csv')


```

- Impressive use of matplotlib to make clean code:

```python

ax = train.plot(x = 'ds', y = 'y', color = 'black', label = 'Train', figsize = (16, 4))
test.plot(x='ds', y='y', color='grey', label='Test', ax=ax)
plt.title('Uber searches throughout the years');
```

**Where you could of improved on:**

-Cleaner way of importing data shown below:

```python 
  
project = 'en.wikipedia'
access = 'all-access'
agent = 'user'
article = 'Veganism'
granularity = 'daily'
start = '20150701'
end = '20190527'

endpoint = f'/metrics/pageviews/per-article/{project}/{access}/{agent}/{article}/{granularity}/{start}/{end}'
url = 'https://wikimedia.org/api/rest_v1' + endpoint
response = requests.get(url)
assert response.status_code == 200
df = pd.DataFrame(response.json()['items'])
df = df.rename(columns={'timestamp':'ds', 'views':'y'})
df['ds'] = pd.to_datetime(df['ds'], format='%Y%m%d%H')

```

# Part 2 - Baselines
***
**What went well:**

- Graphs are beutiful and well done .:

```python
#Nice Indexing
ax = train.plot(x = 'ds', y = 'y', color = 'black', label = 'Train', figsize = (16,4))
test.plot(x = 'ds', y = 'y', color = 'red', label = 'Test', ax=ax)
train.plot(x = 'ds', y = 'naive_baseline', label = 'Naive Baseline, Train', color = 'gold', ax=ax)
test.plot(x = 'ds', y = 'naive_baseline', label = 'Naive Baseline, Test', color = 'Green', ax=ax)
plt.title('Uber searches throughout the years');


ax = train.plot(x='ds', y='y', color='blue', label='Train', figsize=(16,4))
test.plot(x='ds', y='y', color='red', label='Test', ax=ax)
train_forecast.plot(x='ds', y='yhat', color='black', 
                    label='Prophet forecast, train', ax=ax)
test_forecast.plot(x='ds', y='yhat', color='green', 
                   label='Prophet forecast, test', ax=ax)
plt.title('Uber searches throughout the years');

from fbprophet.plot import plot_cross_validation_metric
fig = plot_cross_validation_metric(df_cv, metric='mae')


```


# What to work on
- [x] Comment more in your code!
- [x] Come to PM for questions or help

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
# evaluating_forecasts - "Daniel Calimayor-Montero"

# Part 1 - Loading Data/Analysis 
***
**What went well:**

- Good job with importing your pandas data frame from the csv file: 

```python

u = pd.read_csv('pageviews-20150701-20190523.csv')

```

- Impressive use of matplotlib to make clean code:

```python
ax = train.plot(x = 'ds', y = 'y', color = 'black', label = 'Train', figsize = (16, 4))
test.plot(x='ds', y='y', color='grey', label='Test', ax=ax)
plt.title('Uber searches throughout the years');
```

**Where you could of improved on:**

- uber variable is not needed here and may make things more confusing. u as a variable name could be confusing to new eyes.. df is standard:
```python 

u = pd.read_csv('pageviews-20150701-20190523.csv')

uber = u.rename(columns={'Date':'ds', 'Uber':'y'})
```

# Part 2 - Graphing 
***
**What went well:**

- Nice use of splicing .:

```python
train = uber['2015-07-01':'2019-05-23']
test = uber['2018-05-24': '2019-05-24']



```

- Through the years and perfect code execution! .:

```python
ax = train.plot(x='ds', y='y', color='blue', label='Train', figsize=(16,4))
test.plot(x='ds', y='y', color='red', label='Test', ax=ax)
train_forecast.plot(x='ds', y='yhat', color='black', 
                    label='Prophet forecast, train', ax=ax)
test_forecast.plot(x='ds', y='yhat', color='green', 
                   label='Prophet forecast, test', ax=ax)
plt.title('Uber searches throughout the years');

```
- Great use of Prophet library and using the metrics .:

```python
from fbprophet.plot import plot_cross_validation_metric
fig = plot_cross_validation_metric(df_cv, metric='mae')


```

# What to work on
- [x] Comment more in code 
- [x] Come to PM for questions or help

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
# doing_linear_regression - "Daniel Calimayor"

# Part 1 - Loading Data/Analysis findings
***
**What went well:**

- Great job collecting and grouping data: 

```python

votes = pd.DataFrame(data=data, columns=columns)

growth = pd.DataFrame(data=data, columns=columns)

deaths = pd.DataFrame(data=data, columns=columns)

pay = pd.read_csv('PAYEMS.csv')



```

- Simplistic way of finding target mean. Great job!: 

```python

df['Mean Baseline'] = df[target].mean()
df['Error'] = df['Mean Baseline'] - df[target]
df
```


- Impressive work on finding Absolute Error and mean from sklearn:

```python

df['Absolute Error'] = df['Error'].abs()
df['Absolute Error'].sum()


df['Absolute Error'].mean()

from sklearn.metrics import mean_absolute_error
mean_absolute_error(y_true = df[target], y_pred = df['Mean Baseline'])


```

**Where you could of improved on:**

- I'd try to make more comments: 


```python 


```

# Part 2 - Graphing 
***
**What went well:**

- Code could be be shortened, but looks good! :

```python

ax = df.plot(x = 'Year', y = 'Linear Regression w/ Two Features', 
             label = 'Incumbent Party Vote Share Prediction', color = 'red',
             kind = 'scatter', figsize=(19,10))

df.plot(x = 'Year', y = 'Incumbent Party Vote Share', 
       label = 'Incumbent Party Vote Share Actual', color= 'Purple',
        kind = 'scatter', ax=ax)

ax1 = ax.twinx() 

ax1 = df.plot(x = 'Year', y = 'Absolute Error', label = 'MAE' , color = 'black',
             kind = 'line',figsize=(19,10), ax=ax1)
ax1.set_ylabel('MAE');
ax.legend(frameon=True, loc='best', bbox_to_anchor=(0.5, -0.06), ncol=2)
ax1.legend(frameon=False,loc='best',shadow=True, bbox_to_anchor=(.8, -0.06))

```

# What to work on
- [x] Comment more in code 
- [x] Less lines and more organized notebooks 
- [x] Come to PM for questions or help

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
# Daniel Calimayor-Montero- "Regression 1 Sprint Challenge"

# Part 1. Predict avocado sales
***
**What went well:**

-Good job imporing data and dataset:

```python


import pandas as pd
avocado_url = 'https://drive.google.com/uc?export=download&id=1ljE-LhCA9CCFvSIJYJ1ewX8JmdDpfuhi'
avocado = pd.read_csv(avocado_url)
assert avocado.shape == (169, 2)



```

## 1.2. Plot the forecast

- Impressive use of forcast and knowledge of Prophet!:
```python



fig = m.plot(forecast)

fig1 = m.plot_components(forecast)

avocado_cv.describe(exclude = 'number')

```




```

##1.5. Plot the cross-validation Mean Absolute Percentage Error: metric='mape' 
***
**What went well:**

- Clean and elegant: code. I'm impressed :) :

```python




from fbprophet.plot import plot_cross_validation_metric
fig2 = plot_cross_validation_metric(avocado_cv, metric = 'mae')




```

#Stretch Goal (PARTIAL)

# Part 2 

##2.1. Begin with baselines for regression. Calculate the mean absolute error and 𝑅2 score for a mean baseline.

- Went went the base .head: 
``` python
rent.head()

```

**What could be better

-use this syntax for better insight:
```python

rent.describe()

```


## 2.2. Do train/test split. Use data from April & May 2016 to train. Use data from June 2016 to test.
-Good job choosing features and getting to it (I added another way as well)):
```python


features = ['bedrooms']
target = 'price'
x = rent[features]
y = rent[target]


## Other option

train = rent.query('month < 6')
test = rent.query('month == 6')
train.shape, test.shape

features = ['bedrooms', 'bathrooms']
target = 'price'

```

##2.3. Choose any two features.

-use this syntax for better insight:
```python

```

2.4. Use scikit-learn to fit a Linear Regression model on the train data.

-Not bad for just starting out to use to scikit. Syntax is not bad, but features could be better ( see below ):
```python

from sklearn.metrics import mean_absolute_error, r2_score
import matplotlib.pyplot as plt

def plot_preds(x, y, y_pred):
    plt.scatter(x, y, label='y_true')
    plt.plot(x, y_pred, label='y_pred')
    plt.legend()
    
    
    mae = mean_absolute_error(y, y_pred)
    r2 = r2_score(y, y_pred)
    print('Mean Absolute Error:', mae)
    print('R^2:', r2)
    
    
    
plot_preds(x, y, y_pred)
plt.title('Mean Baseline');

```

**What could be better

-better syntax:


```python

# Choose a class of model by importing the appropriate estimator class from Scikit-Learn.
from sklearn.linear_model import LinearRegression

# Choose model hyperparameters by instantiating this class with desired values.
model = LinearRegression()

# Arrange data into a features matrix and target vector.
X_train = train[features]
y_train = train[target]
X_test = test[features]
y_test = test[target]

X_train.shape, y_train.shape, X_test.shape, y_test.shape

# Fit the model to your data by calling the fit() method of the model instance.
model.fit(X_train, y_train)

# Apply the Model to new data: For supervised learning, often we predict labels for unknown data using the predict() method.
y_pred = model.predict(X_test)

```

2.6. Get regression metrics MAE and 𝑅2 for the test data.

-Great job:
```python


mae = mean_absolute_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)
print('Test Mean Absolute Error:', mae)
print('Test R^2 Score:', r2)

```

2.7. Get the model's coefficients and intercept.

https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html

-use this syntax for better insight:
```python
model.intercept_, model.coef_
```

#Stretch Goals_2

```python

import seaborn as sns

target = 'price'
features = ['bedrooms', 'bathrooms', 'latitude', 'longitude']
for feature in features:
    sns.lmplot(x=feature, y=target, data=train, scatter_kws={'alpha':0.01})
    
    ```
    
    
    
    ```python
    
    
    def fit(train, test, features):
    X_train = train[features]
    y_train = train[target]
    X_test = test[features]
    y_test = test[target]

    model = LinearRegression()
    model.fit(X_train, y_train)

    print(f'Linear Regression with {len(features)} features: {features}')
    print('Train Mean Absolute Error:', 
          mean_absolute_error(y_train, model.predict(X_train)))
    print('Train R^2 Score:', 
          r2_score(y_train, model.predict(X_train)))
    print('Test Mean Absolute Error:', 
          mean_absolute_error(y_test, model.predict(X_test)))
    print('Test R^2 Score:', 
          r2_score(y_test, model.predict(X_test)))

features = ['bedrooms', 'bathrooms', 'latitude', 'longitude']
fit(train, test, features)
```

-with 3 features:

```python
features = ['bedrooms', 'bathrooms', 'longitude']
fit(train, test, features)
```

# What to work on
- [x] Comment more in code 
- [x] Less lines and more organized notebooks 
- [x] Come to PM for questions or help

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
EDIT: MY MISTAKE DID NOT SEE FILE WAS OUTSIDE OF FOLDER

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
# Daniel Calimayor-Montero- Regression 1 Sprint Challenge

# Part 1. Predict avocado sales 

***
**What went well:**

-Good collective use of imports and reading in data of the :

```python

import pandas as pd
avocado_url = 'https://drive.google.com/uc?export=download&id=1ljE-LhCA9CCFvSIJYJ1ewX8JmdDpfuhi'
avocado = pd.read_csv(avocado_url)
assert avocado.shape == (169, 2)

```
#Part 2. Predict NYC apartment rent 

##imports


- Nice import and assert at line 5: 

```python



import pandas as pd
rent_url = 'https://drive.google.com/uc?export=download&id=1lVVPmh-WYGb_0Gs_lOv22i02-150qDdZ'
rent = pd.read_csv(rent_url)
assert rent.shape == (48295, 7)



```
- Simple and clean code for MAE. Well Done. :

```python

m = 2
b = 15
y_pred = m*x + b
plot_preds(x, y, y_pred)
plt.title('Guessing & Checking by hand');

```

# What to work on
- [x] Comment more in code
- [x] Come to PM for questions or help
